django/db/models/sql/query.py

```diff
@@ -1021,6 +1021,17 @@ def change_aliases(self, change_map):
                 if alias == old_alias:
                     table_aliases[pos] = new_alias
                     break
+
+        # 3. Rename the external aliases of combined queries.
+        for combined_query in self.combined_queries:
+            external_change_map = {
+                alias: aliased
+                for alias, aliased in change_map.items()
+                if alias in combined_query.external_aliases
+            }
+            if external_change_map:
+                combined_query.change_aliases(external_change_map)
```

Comment:
`change_aliases()` is already no-op for empty maps, so I would skip this check:
```suggestion
            combined_query.change_aliases(external_change_map)
```

Reply:
Thanks for the review @felixxm as usual 🙇 🏅 I forgot about f3d10546a850df4fe3796f972d5b7e16adf52f54.